src/java/org/apache/cassandra/locator/InetAddressAndPort.java

```diff
@@ -159,6 +163,29 @@ public static InetAddressAndPort getByName(String name) throws UnknownHostExcept
         return getByNameOverrideDefaults(name, null);
     }
 
+
+    public static List<InetAddressAndPort> getAllByName(String name) throws UnknownHostException
+    {
+        return getAllByNameOverrideDefaults(name, null);
+    }
+
+    /**
+     *
+     * @param name Hostname + optional ports string
+     * @param port Port to connect on, overridden by values in hostname string, defaults to DatabaseDescriptor default if not specified anywhere.
+     */
+    public static List<InetAddressAndPort> getAllByNameOverrideDefaults(String name, Integer port) throws UnknownHostException
+    {
+        HostAndPort hap = HostAndPort.fromString(name);
+        if (hap.hasPort())
+        {
+            port = hap.getPort();
+        }
+        Integer finalPort = port;
+
+        return Stream.of(InetAddress.getAllByName(hap.getHost())).map((address) -> getByAddressOverrideDefaults(address, finalPort)).collect(Collectors.toList());
```

Comment:
Nit: we could break this line
```suggestion
        return Stream.of(InetAddress.getAllByName(hap.getHost()))
                     .map((address) -> getByAddressOverrideDefaults(address, finalPort))
                     .collect(Collectors.toList());
```